src/metadata/utils.ts

```diff
@@ -57,3 +59,10 @@ export function ensureReflectMetadataExists() {
     throw new ReflectMetadataMissingError();
   }
 }
+
+export function flattenExtensions(
```

Comment:
I'm not such a fan of having a "util" just for this but I didn't see a proper alternative.

I could make it more generic by accepting a key by which we would flatten instead of hardcoding extensions, but since there is no other use through the codebase now it looked like premature abstraction to me